staging/src/k8s.io/cli-runtime/pkg/resource/helper.go

```diff
@@ -96,20 +100,28 @@ func (m *Helper) WithSubresource(subresource string) *Helper {
 	return m
 }
 
+// WithWarningHandler sets the custom warning handler
+func (m *Helper) WithWarningHandler(warningHandler rest.WarningHandler) *Helper {
+	m.WarningHandler = warningHandler
+	return m
+}
+
 func (m *Helper) Get(namespace, name string) (runtime.Object, error) {
 	req := m.RESTClient.Get().
 		NamespaceIfScoped(namespace, m.NamespaceScoped).
 		Resource(m.Resource).
 		Name(name).
-		SubResource(m.Subresource)
+		SubResource(m.Subresource).
+		WarningHandler(m.WarningHandler)
```

Comment:
RESTClient already has a warning handler
https://github.com/kubernetes/kubernetes/blob/ae1caa40a29c5476efe51e8ce2ce33812e1ce8ce/staging/src/k8s.io/client-go/rest/client.go#L97-L99
which it propagates into the requests it creates:
https://github.com/kubernetes/kubernetes/blob/ae1caa40a29c5476efe51e8ce2ce33812e1ce8ce/staging/src/k8s.io/client-go/rest/request.go#L163

Constructing the RESTClient with the desired warning handler will propagate that handler into the requests.

I'm not sure I understand why this PR is needed.

In fact, if someone was already constructing their RESTClient with a warning handler and successfully using it here, this PR would break them (calling `WarningHandler(m.WarningHandler)` when `m.WarningHandler` is nil, as it is by default, will erase the warning handler that came from the rest client)

Reply:
Thank you for review. It makes sense. 
/close